src/main/java/steps/NotePageSteps.java

```diff
@@ -1,8 +1,106 @@
 package steps;
 
+import business_objects.CheckListNote;
 import business_objects.Note;
+import business_objects.TextNote;
 import pages.BasePage;
+import pages.ChecklistNotePage;
+import pages.NotePage;
+import pages.TextNotePage;
+import popups.ColorPopup;
+import popups.NewChecklistItemPopup;
 
-public abstract class NotePageSteps extends BasePage {
-    public abstract void setNoteData(Note note);
+import java.util.Iterator;
+import java.util.List;
+
+public class NotePageSteps extends BasePage {
+
+    public void setNoteData(Note note) {
+        switch (note.getType()) {
+            case TEXT_NOTE: {
+                TextNote textNote = (TextNote) note;
+                Iterator iterator = textNote.getBody().getContentItems().iterator();
+                String body = "";
+                while (iterator.hasNext()){
+                    body += iterator.next();
+                }
+                if (body != null) {
+                    new TextNotePage().setBody(body);
+                }
+                break;
+            }
+            case CHECKLIST: {
+                CheckListNote checkListNote = (CheckListNote) note;
+                List<String> itemsList = checkListNote.getBody().getContentItems();
+                if (itemsList != null) {
+                    itemsList
+                            .forEach(item -> {
+                                new ChecklistNotePage().clickNewItem();
+                                NewChecklistItemPopup popup = new NewChecklistItemPopup();
+                                popup.setItemBody(item);
+                                popup.clickOkButton();
+                            });
+                }
+                break;
+            }
+            default:
+                throw new RuntimeException("This note type doesn't exists!");
+        }
+    }
+
+    public void setTitle(Note note) {
+        NotePage notePage = new NotePage();
+        notePage.setTitle(note.getName());
+    }
+
+    public void setColor(Note note) {
```

Comment:
Split this method into two separate methods: 
clickColorButton() 
& 
setColor(Color color)

Reply:
done